src/axom/core/Array.hpp

```diff
@@ -926,20 +1016,23 @@ template <typename T, int DIM, MemorySpace SPACE>
 inline void Array<T, DIM, SPACE>::swap(Array<T, DIM, SPACE>& other)
 {
   ArrayBase<T, DIM, Array<T, DIM, SPACE>>::swap(other);
-  T* temp_data = m_data;
-  IndexType temp_num_elements = m_num_elements;
-  IndexType temp_capacity = m_capacity;
-  double temp_resize_ratio = m_resize_ratio;
+  T* const temp_data = m_data;
+  const IndexType temp_num_elements = m_num_elements;
+  const IndexType temp_capacity = m_capacity;
+  const double temp_resize_ratio = m_resize_ratio;
+  const bool temp_default_construct = m_default_construct;
 
   m_data = other.m_data;
   m_num_elements = other.m_num_elements;
   m_capacity = other.m_capacity;
   m_resize_ratio = other.m_resize_ratio;
+  m_default_construct = other.m_default_construct;
 
   other.m_data = temp_data;
   other.m_num_elements = temp_num_elements;
   other.m_capacity = temp_capacity;
   other.m_resize_ratio = temp_resize_ratio;
+  other.m_default_construct = temp_default_construct;
```

Comment:
Minor: Any reason not to implement these in terms of `axom::swap()` ?

It should be effectively the same, but might read a bit easier?

See: https://github.com/LLNL/axom/blob/f18c0decad8ea1d918acb50aba9a1a11bb0bcf4b/src/axom/core/utilities/Utilities.hpp#L74-L86

Reply:
This is a good point - I think I had done it like this to stay consistent but there shouldn't be any reason we can't just swap all the members with axom::swap